### PR TITLE
[@types/express-serve-static-core] Add link and unlink method for Express Router

### PR DESCRIPTION
**Pull request by quangtran88**
Add missing `link` and `unlink` method for Express Router

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
* https://www.ietf.org/rfc/rfc2068.txt, missing LINK and UNLINK method https://github.com/jshttp/methods/issues/16
* Underneath Express Router using package `methods`: [Source code](https://github.com/expressjs/express/blob/0debedf4f31bb20203da0534719b9b10d6ac9a29/lib/router/route.js#LL206C1-L206C8)
* Package `methods` using `METHODS` from nodejs `http` package: [Source code](https://github.com/jshttp/methods/blob/90dd6f9a046fffaf158287cb19eee686d17e7cde/index.js#L30)
* NodeJS has already supported these two HTTP method **LINK** and **UNLINK** since the first LTS `4.9.1` :
```
❯ node -v
v4.9.1
❯ node   
> require('http').METHODS
[ 'ACL',
  'BIND',
  'CHECKOUT',
  'CONNECT',
  'COPY',
  'DELETE',
  'GET',
  'HEAD',
  'LINK',
  'LOCK',
  'M-SEARCH',
  'MERGE',
  'MKACTIVITY',
  'MKCALENDAR',
  'MKCOL',
  'MOVE',
  'NOTIFY',
  'OPTIONS',
  'PATCH',
  'POST',
  'PROPFIND',
  'PROPPATCH',
  'PURGE',
  'PUT',
  'REBIND',
  'REPORT',
  'SEARCH',
  'SOURCE',
  'SUBSCRIBE',
  'TRACE',
  'UNBIND',
  'UNLINK',
  'UNLOCK',
  'UNSUBSCRIBE' ]
```
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by gabritto** [comment]
I can't see how the RFC and issue links you provided relate to the type definitions in question, since I'm not an expert on these types or the original project. You can either try to explain it here in more details, and maybe I or some other DT maintainer will be able to review this change, or you can wait for an owner of these type definitions to review.

**Review by andrewbranch — CHANGES_REQUESTED** [review]
Echoing what Gabriela said, please provide a link to either docs or source code (as opposed to discussion) that proves the existence of these methods.

**Comment by quangtran88** [comment]
Hi @andrewbranch, documentation and source code which provides context for the suggested changes are updated.

**Review by andrewbranch — APPROVED** [review]

**Comment by quangtran88** [comment]
Ready to merge

**Pull request merged**
